experimental/ssh/internal/proxy/proxy.go

```diff
@@ -170,14 +172,12 @@ func (pc *proxyConnection) signalClosedConnection(err error) error {
 	}
 }
 
-func (pc *proxyConnection) blockUntilHandoverComplete(conn *websocket.Conn) error {
-	pc.handoverMutex.Lock()
-	defer pc.handoverMutex.Unlock()
-	// Sanity check to ensure we are not in the middle of a handover - this should not happen.
-	if pc.conn.Load() == conn {
-		return errors.New("handover mutex acquired while the old connection is still active")
+func (pc *proxyConnection) waitForNewConnection(conn *websocket.Conn) {
+	pc.connChanged.L.Lock()
+	defer pc.connChanged.L.Unlock()
+	for pc.conn.Load() == conn {
```

Comment:
is there a possibility that this loop would wait for indefinite time? if so, shall we add a waiting timeout and fail?